### DISCUSSION TITLE
Workspace is unhealthy _Your workspace is running but 1 agent is unhealthy

### DISCUSSION THREAD

**Discussion by Chen-0522** [coder/coder#15491] [General]
Version: [`v2.8.2+c4a01a4`](https://github.com/coder/coder/commit/c4a01a42ceeb8857856c2204e4d4907f0f93ae5f)![Image](https://github.com/user-attachments/assets/5b04d035-cdcf-410f-a3d5-6ceac04728ea)
![Image](https://github.com/user-attachments/assets/3fe3ec2c-eb1b-4c56-a3a2-bd05da6cab8d)

**Comment by matifali** [comment]
Hi @Chen-0522 could you try updating the Coder server to the latest version? 2.16.0 or 2.15.3?

Also please exec/ssh into your workspace outside coder and see if the workspace can reach the Coder server. 

Do you have https set up on your coder server?

**Comment by dannykopping** [comment]
@Chen-0522 did the above solve your issue?

**Comment by eadoking** [comment]
any body was able to solve this issue ? I have it also, and upgraded to v2.16.0+683a720 same issue 

![Image](https://github.com/user-attachments/assets/96fc9140-4608-4e28-b809-e6dec68eea7f)
![Image](https://github.com/user-attachments/assets/206bd987-e5f4-4aad-bb48-edf181af339e)
![Image](https://github.com/user-attachments/assets/407f83c3-ade2-4e8b-ae9e-ea1cf416ca82)
![Image](https://github.com/user-attachments/assets/ed052f33-5e22-4717-a27d-e7d5406e6dc8)

I am exposing the coder directly to server ip on :80 and 443 +zerossl certificate but no success

system debian 12. nftables disabled.

`/ coder.env

CODER_ACCESS_URL=https://xyzpoc.dev/
CODER_WILDCARD_ACCESS_URL=*.xyzpoc.app
CODER_HTTP_ADDRESS=1.2.3.4:80
CODER_DISABLE_DIRECT_CONNECTIONS=true
CODER_TLS_ADDRESS=1.2.3.4:443
CODER_REDIRECT_TO_ACCESS_URL=true

CODER_TLS_CERT_FILE=/etc/coder/fullchain.cer
CODER_TLS_ENABLE=true
CODER_TLS_KEY_FILE=/etc/coder/cert.key

CODER_VERBOSE=true


CODER_STRICT_TRANSPORT_SECURITY=1
CODER_TELEMETRY_ENABLE=false
CODER_DISABLE_NETWORK_TELEMETRY=true
CODER_TUNNEL=false
CODER_TLS_MIN_VERSION=tls12
CODER_DERP_SERVER_ENABLE=true
CODER_DERP_SERVER_REGION_NAME=DE
CODER_DERP_FORCE_WEBSOCKETS=true

CODER_PG_CONNECTION_URL="postgresql://user:password@127.0.0.1:5432/coderdb?sslmode=disable"`

**Comment by matifali** [comment]
Hi @eadoking if you exec into the workspace compute and see if it can reach the access url?

Also is there something that is failing in your `startup_script`?

**Comment by eadoking** [comment]
Hi @matifali, thanks for reply 

Actually, no can't, and no errors in `startup_script`. just dead-silence.  I tried to check the human log, also not written.


I tried couple times using `curl -L https://coder.com/install.sh | sh` to install coder on Debian 12, always got the loading state and then unhealthy.

I tried the server with new image and nothing installed only coder, the same. (debian 12)

I tried to expose it directly to 443, and tried with nginx .. always no load.

## how it worked at the end

only worked when I used alternative methods 

`https://coder.com/docs/install/docker` 

so, I installed docker, and copied the docker-compose file from your github repo and then added a debian service to run the docker composer up for coder,
this way, it worked and now I have a runing workspaces and template.

```
[Unit]
Description=Docker Compose coder
After=network.target docker.service
Requires=docker.service

[Service]
WorkingDirectory=/home/admin/.config/coderv2-docker
ExecStart=/usr/bin/docker compose up
ExecStop=/usr/bin/docker compose down
Restart=always
Type=simple

[Install]
WantedBy=multi-user.target
```

**Comment by matifali** [comment]
That's interesting.

Also if you are using docker compose you probably don't need to create a service as docker compose can manage the lifecycle of services it runs on its own.

**Comment by PecTheFabricator** [comment]
@matifali sorry for jumping in this thread. However I am facing the same issue. Basically, if I use this image https://quay.io/repository/jupyter/base-notebook the container start, but coder space can't access the container for some reason. However, I am able to exec the container from the docker host. I have tried as pod and as container, not success at all.

**Reply by matifali** [reply]
Hi, @PecTheFabricator, could you move to a new discussion, and I can help troubleshoot the issue? Thanks.

**Comment by PecTheFabricator** [comment]
> [@matifali](https://github.com/matifali) sorry for jumping in this thread. However I am facing the same issue. Basically, if I use this image https://quay.io/repository/jupyter/base-notebook the container start, but coder space can't access the container for some reason. However, I am able to exec the container from the docker host. I have tried as pod and as container, not success at all.

Just adding

I found that the code server is not being dowloaded to the container. May be that is the issue

**Comment by homjay** [comment]
I have the same problem; the workspace is unable to boot.
